drivers/display/ls0xx.c

```diff
@@ -0,0 +1,501 @@
+		k_msleep(500);
+	}
+}
+
+/* Driver will handle VCOM toggling */
+K_THREAD_STACK_DEFINE(vcom_toggle_stack, 256);
+struct k_thread vcom_toggle_thread;
+#endif /* CONFIG_LS0XX_VCOM_EXTERNAL */
+
+#else
+#define LS0XX_EXTCOMIN_BIT 0x00
+#endif /* CONFIG_LS0XX_VCOM_DRIVER */
+
+/**
+ * @brief Display memory contents of the display
+ *
+ * @details Display memory contents, ie actual image data
+ *          During initialization DISP pin will be set high
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_blanking_off(const struct device *dev)
+{
+#if defined(LS0XX_DISP_CNTRL)
+	struct ls0xx_data *driver = dev->data;
+
+	return gpio_pin_set(driver->disp, LS0XX_DISP_PIN, 1);
+#else
+	LOG_ERR("Unsupported");
+	return -ENOTSUP;
+#endif
+}
+
+/**
+ * @brief Display becomes completely white
+ *
+ * @details Displays complete white while but the
+ *          memory contents are retained
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_blanking_on(const struct device *dev)
+{
+#if defined(LS0XX_DISP_CNTRL)
+	struct ls0xx_data *driver = dev->data;
+
+	return gpio_pin_set(driver->disp, LS0XX_DISP_PIN, 0);
+#else
+	LOG_ERR("Unsupported");
+	return -ENOTSUP;
+#endif
+}
+
+/**
+ * @brief Clear the internal memory of the display
+ *
+ * @details Clear the display memory and the frame_buf
+ *          setting it to all white
+ *
+ * @param dev Pointer to the display device
+ */
+static int ls0xx_clear(const struct device *dev)
+{
+	struct ls0xx_data *driver = dev->data;
+	int err;
+	uint8_t clear_cmd[2] = {LS0XX_BIT_CLEAR | LS0XX_EXTCOMIN_BIT, 0};
+
+	tx_buf.len = sizeof(uint16_t);
+	tx_buf.buf = &clear_cmd;
+
+	gpio_pin_set(driver->scs, LS0XX_SCS_PIN, 1);
+	err = spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+	gpio_pin_set(driver->scs, LS0XX_SCS_PIN, 0);
+
+	return err;
+}
+
+/**
+ * @brief Send required lines to the LCD
+ *
+ * @details Update the display by sending num_lines
+ *          starting from start_line.
+ *
+ * @param dev Pointer to the display device
+ * @param start_line Starting line number, starts with 1
+ *                   corresponds to Y co-ordinate + 1
+ * @param num_lines Number of lines ie height to update
+ * @param buf  Buffer containing data to write
+ */
+#if defined(__GNUC__)
+	__attribute__((optimize("O3")))
+#endif
+static int ls0xx_update_display(const struct device *dev,
+								uint16_t start_line,
+								uint16_t num_lines,
+								const uint8_t *buf)
+{
+	struct ls0xx_data *driver = dev->data;
+
+	int err;
+	uint8_t write_cmd;
+	uint16_t currentline;
+
+	LOG_DBG("Lines %d to %d", start_line, start_line+num_lines-1);
+
+	gpio_pin_set(driver->scs, LS0XX_SCS_PIN, 1);
+
+	write_cmd = LS0XX_BIT_WRITECMD | LS0XX_EXTCOMIN_BIT;
+	extcomin_bit = LS0XX_EXTCOMIN_BIT ? 0x00 : LS0XX_BIT_VCOM;
+	tx_buf.len = sizeof(uint8_t);
+	tx_buf.buf =  &write_cmd;
+	err = spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
+
+	if (buf != NULL) {
```

Comment:
just error and exit at the top if no buffer is provided.

Reply:
We need to send the VCOM toggle command